rotkehlchen/api/rest.py

```diff
@@ -4435,3 +4442,86 @@ def export_history_events(
 
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    def add_accounting_rule(
+            self,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            counterparty: Optional[str],
+            rule: 'BaseEventSettings',
+    ) -> Response:
+        db = DBAccountingRules(self.rotkehlchen.data.db)
+        try:
+            db.add_accounting_rule(
+                event_type=event_type,
+                event_subtype=event_subtype,
+                counterparty=counterparty,
+                rule=rule,
+            )
+        except InputError as e:
+            return api_response(wrap_in_fail_result(str(e)), status_code=HTTPStatus.CONFLICT)
+
+        return api_response(_wrap_in_ok_result(True), status_code=HTTPStatus.OK)
+
+    def update_accounting_rule(
+            self,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            counterparty: Optional[str],
+            rule: 'BaseEventSettings',
+            row_id: int,
+    ) -> Response:
+        db = DBAccountingRules(self.rotkehlchen.data.db)
+        try:
+            db.update_accounting_rule(
+                event_type=event_type,
+                event_subtype=event_subtype,
+                counterparty=counterparty,
+                rule=rule,
+                row_id=row_id,
+            )
+        except InputError as e:
+            return api_response(wrap_in_fail_result(str(e)), status_code=HTTPStatus.CONFLICT)
+
+        return api_response(_wrap_in_ok_result(True), status_code=HTTPStatus.OK)
+
+    def delete_accounting_rule(
+            self,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            counterparty: Optional[str],
+    ) -> Response:
+        db = DBAccountingRules(self.rotkehlchen.data.db)
+        try:
+            db.remove_accounting_rule(
+                event_type=event_type,
+                event_subtype=event_subtype,
+                counterparty=counterparty,
+            )
+        except InputError as e:
+            return api_response(wrap_in_fail_result(str(e)), status_code=HTTPStatus.CONFLICT)
+
+        return api_response(_wrap_in_ok_result(True), status_code=HTTPStatus.OK)
+
+    def query_accounting_rules(self, filter_query: AccountingRulesFilterQuery) -> Response:
+        db = DBAccountingRules(self.rotkehlchen.data.db)
+        entries, total_filter_count = db.query_rules(filter_query=filter_query)
+        entries_result = []
+        for entry in entries:
+            # serialize the rule and add information about the key to what it applies
+            data = entry[1].serialize()
+            data['rowid'] = entry[0][0]
+            data['event_type'] = entry[0][1]
+            data['event_subtype'] = entry[0][2]
+            data['counterparty'] = entry[0][3] if entry[0][3] != NULL_ACCOUNTING_COUNTERPARTY else None  # noqa: E501
```

Comment:
hmmm why does this for loop serialization happen here? Can't you have it as part of this query_rules thing or somehow a different method?

Reply:
Made it here since the method was more a db operation only but I've added a new method that includes serialization